### PR TITLE
feat: Make it possible to keybind `TypableCommands`

### PR DESCRIPTION
**Pull request by Nehliin**
Fixes: https://github.com/helix-editor/helix/issues/852

This makes it possible to add keymaps for `TypableCommands` (including aliases + commands with arguments) in your `config.toml` like the following example:
```toml
[keys.normal]
"space" = { "t" = ":earlier 5 sec", "y" = ":theme default"}
"f"  = ":format"
"C-s" = ":w"
# Previously mapped "static commands"
"C-l" = "jump_view_right"
"C-h" = "jump_view_left"
"C-j" = "jump_view_down"
"C-k" = "jump_view_up"
```

This solution is a lot less hacky than my previous pr and makes it possible to map all current and future `TypableCommands`. I decided to not go for the `[ .. ]` surround for command arguments because it would just be more of a hassle to parse with little benefit for users imo. Arguments are passed by space separating them after the initial command instead. 

**Update: aliases are supported as well so updated the example**

### CODE REVIEW

File: helix-term/src/commands.rs
Comment (sudormrfbin) [inline helix-term/src/commands.rs:176]: `COMMANDS` is a confusing name now with typable command and static commands, renaming it would be nice (`TYPABLE_COMMAND_LIST` is already taken though).

File: helix-term/src/commands.rs
Comment (sudormrfbin) [inline helix-term/src/commands.rs:5708]: This should be `prompt` ?

Comment (Nehliin) [reply helix-term/src/commands.rs:5708]: Whoops yeah thats just a left over from when I tested the rename symbol command :)

File: helix-term/src/commands.rs
Comment (Nehliin) [reply helix-term/src/commands.rs:417]: ah nice missed that, thanks!

File: helix-term/src/commands.rs
Comment (sudormrfbin) [inline helix-term/src/commands.rs:1891]: Why was the context removed ? Borrow checker problems ?

Comment (Nehliin) [reply helix-term/src/commands.rs:1891]: The context provided by the `execute` method in the `MappableCommand` is the `Context<'a>`  struct defined within the `command.rs` module. This method was previously used only for static commands but is now used to call both types of command. The context used previously for the `TypableCommands` is actually the `compositor::Context` which is a different type all together. They both include `Editor` and `Jobs` which are the two things all of the `TypableCommands` require so I simply just replaced the compositor Context with the actual fields needed.

File: helix-term/src/commands.rs
Comment (Nehliin) [inline helix-term/src/commands.rs:447]: @sudormrfbin  this works but I'm not sure if we want to separate the Static commands from Typable commands here for the future or if we should just keep it as is for now.

File: helix-term/src/commands.rs
Comment (archseer) [reply helix-term/src/commands.rs:1891]: I'd rather see that you keep the signature and just construct a compositor::Context

File: helix-term/src/commands.rs
Comment (sudormrfbin) [reply helix-term/src/commands.rs:447]: I think it's better to check for static and typable here since they are different under the hood anyways.

**Comment by Nehliin** [comment]
Rebased updated `PartialEq` implementation and reverted to `compositor::Context` now

**Comment by archseer** [comment]
Yeah, I've been wanting to do this for a while but it will break pending PRs in the queue. Let's wait until I manage to catch up with the merges.

**Comment by Nehliin** [comment]
@archseer Sure, like I said I don't want to do that in this pr since that would make it quite huge. Better to do it in a separate pr imo

**Review by archseer — APPROVED** [review]
LGTM but I'd like to get an approval from @sudormrfbin too :)

**Review by sudormrfbin — APPROVED** [review]
LGTM ! Great work @Nehliin 🎉

**Pull request merged**
